python/samba/samba3/__init__.py

```diff
@@ -80,20 +79,20 @@ def __len__(self):
 
     def keys(self):
         """Return list with all the keys."""
-        return [k.rstrip("\x00") for k in self.db.iterkeys() if not k.startswith(REGISTRY_VALUE_PREFIX)]
+        return [k.rstrip(b"\x00") for k in self.db if not k.startswith(REGISTRY_VALUE_PREFIX)]
 
     def subkeys(self, key):
         """Retrieve the subkeys for the specified key.
 
         :param key: Key path.
         :return: list with key names
         """
-        data = self.db.get("%s\x00" % key)
+        data = self.db.get(b"%s\x00" % key)
```

Comment:
% formatting for bytes is added in Python 3.5, formatting like this line will cause error in Python 3.4:
https://www.python.org/dev/peps/pep-0461/

Reply:
well, I was testing against 3.6, wasn't aware of this. However our python3 support is new, this is new code, imo for python3 we should mandate the minumum version (to be something reasonable) that will make life easiest for both porting and making writing it easier to write python2/python3 compatible code. How far back in the python3 version stream will we have to support otherwise ?

besides changes like

-data = self.db.get("%s\x00" % key)
+data = self.db.get(b"%s\x00" % key)

now becoming

data = self.db.get("%s\x00" % key)
+data = self.db.get(key.encode('utf8') + '\x00')

being even uglier than before as the constructs get more complicated (and we are sure to have some of those) it will become even more error prone and harder to easily change. Just my 2c

Reply:
Noel, it is probably best to discuss minimum versions on the list, because almost nobody sees these review comments.

As far as I can tell, the main blocker for >3.4 is the main autobuild machine which runs Ubuntu 14.04 with 3.4.